docusaurus/docs/Flutter/guides/migration_guide_2_0.mdx

```diff
@@ -0,0 +1,299 @@
+
+:::note
+The major changes in `stream_chat_flutter_core` consist of changing over from a map full of options
+to a more type safe and sound approach by changing over to explicit parameters.
+:::
+
+1) Change over Core widget implementations by using the explicit parameters instead of the options map.
+Use these explicit parameters in the widget constructor instead of the option keys:
+
+    * options.state -> bool state
+    * options.watch -> bool watch
+    * options.presence -> bool presence
+
+2) Change over query calls in the BLoCs in the same way (change from options map to explicit parameters
+in the constructor)
+
+### Changelog of `stream_chat`
+
+#### 🛑️ Breaking Changes from 1.5.3
+
+* Migrate this package to null safety
+* ConnectUserWithProvider now requires tokenProvider as a required param. (Removed from the constructor)
+* client.disconnect() is now divided into two different functions
+    * client.closeConnection() -> for closing user websocket connection.
+    * client.disconnectUser() -> for disconnecting user and resetting client state.
+* client.devToken() now returns a Token model instead of String.
+* ApiError is removed in favor of StreamChatError
+    * StreamChatError -> parent type for all the stream errors.
+    * StreamWebSocketError -> for user websocket related errors.
+    * StreamChatNetworkError -> for network related errors.
+* client.queryChannels(), channel.query() options param is removed in favor of individual params
+    * option.state -> bool state
+    * option.watch -> bool watch
+    * option.presence -> bool presence
+* client.queryUsers() options param is removed in favor of individual params
+    * option.presence -> bool presence
+* Added typed filters
+
+#### 🐞 Fixed
+
+* #369: Client does not return without internet connection
+* several minor fixes
+* performance improvements
+
+#### ✅ Added
+
+* New Location enum is introduced for easily changing the client location/baseUrl.
+* New client.openConnection() and client.closeConnection() is introduced to connect/disconnect user ws connection.
+* New client.partialUpdateMessage and channel.partialUpdateMessage methods
+* connectWebSocket parameter in connect user calls to use the client in "connection-less" mode.
+
+#### 🔄 Changed
+
+* baseURL is now deprecated in favor of using Location to change data location.
+
+### Migrating to 2.0 for `stream_chat`
+
+If you are migrating your full Flutter project to null-safety, first make sure you follow the
+instructions from the [official Null Safety migration guide](https://dart.dev/null-safety/migration-guide).
+:::
+
+To migrate to v2.0 for `stream_chat`, first change the version of the package to the latest
+null-safe version.
+
+```yaml
+dependencies:
+  stream_chat: ^2.0.0
+```
+
+Upon doing this, all breaking changes from the package will take immediate effect. Here are steps to
+remedy the issue:
+
+1) Change over the constructor of `connectUserWithProvider()` to the new format.
```

Comment:
This line doesn't make clear what changes need to be made. What is the new format? Add some more detail.